src/client/microsoftAuth.js

```diff
@@ -34,7 +34,9 @@ async function authenticate (client, options) {
     availableProfile: [profile]
   }
   Object.assign(client, certificates)
+  client.profileKeys.expireTime = client.profileKeys.expiresOn.getTime()
   client.session = session
+  client.uuid = profile.id.replace(/(\w{8})(\w{4})(\w{4})(\w{4})(\w{12})/, '$1-$2-$3-$4-$5')
```

Comment:
why are the dashes needed?

Reply:
I just copied the server implementation as I don't know how protodef treats UUIDs

Reply:
Client.uuid should already be defined elsewhere

Reply:
It usually is set after login success packet. So it is not defined before login start